### PR TITLE
Store the gluster volume bricks in order

### PR DESCRIPTION
**Pull request by nnDarshan**
The order of gluster volume bricks is important
because, we figure out the sub-volume details by
seeing the order of bricks. The bricks were not
ordered before, this patch makes sure we save the
bricks in order

tendrl-bug-id: Tendrl/gluster-integration#236
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@shtripat @r0h4n @nthomas-redhat Please have a look

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 1]
```diff
@@ -143,6 +143,18 @@ def _run(self):
                                 ],
                             )
                             NS.volume.save()
+
+                            try:
```
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: you mean every sync we should delete all the bricks and add them back? Why so?

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 2]
```diff
@@ -168,7 +180,20 @@ def _run(self):
                                                 'volume%s.brick%s.mount_options' % (
                                                     index, b_index))
                                         )
-                                    NS.brick.save()
+                                    # Store the bricks sequentially, as the order of bricks
+                                    # matter while figuring out the sub-volumes
+                                    b = NS.brick.__dict__.copy()
+                                    b.pop('_etcd_cls')
+                                    b['name'] = NS.brick.path.replace("/", "_")
+                                    NS.etcd_orm.client.write(
```
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Can we use brick object's save() function here?

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 3]
```diff
@@ -168,7 +180,20 @@ def _run(self):
...
+                                        "clusters/%s/Volumes/%s/Bricks/" % (
+                                            NS.tendrl_context.integration_id,
+                                            NS.volume.vol_id,
+                                        ),
+                                        json.dumps(b),
```
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: How we make sure order is maintained here?

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 1 above)
Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Now we are writing the bricks with "append=true". So each brick will have sequence_no as key and this maintains the order of bricks. If we don't delete it every time it gets accumulated.

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 2 above)
Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/__init__.py]: save method doesn't have the "append" parameter mentioned above

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 3 above)
Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/__init__.py]: As mentioned above, we use the sequence no. to figure out the order

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 1 above)
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Oh somehow I overlooked I feel. Thats fine..

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 2 above)
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: ok

File: tendrl/gluster_integration/sds_sync/__init__.py (hunk 3 above)
Comment (shtripat) [inline tendrl/gluster_integration/sds_sync/__init__.py]: ok

**Comment by shtripat** [comment]
LGTM

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 4]
```diff
@@ -168,7 +180,20 @@ def _run(self):
                                                 'volume%s.brick%s.mount_options' % (
                                                     index, b_index))
                                         )
-                                    NS.brick.save()
+                                    # Store the bricks sequentially, as the order of bricks
+                                    # matter while figuring out the sub-volumes
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Whats the reason to overload NS with NS.brick or NS.volume here? 

If there is no important reason, Id prefer we dont overload NS in this file at all. Please use local variables for brick, volume, vol_opts etc which are assigned NS right now

Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/__init__.py]: Ack. Will change them

**Comment by nnDarshan** [comment]
@r0h4n Have incorporated your suggestions

**Comment by r0h4n** [comment]
Verified?

**Comment by nnDarshan** [comment]
@r0h4n Yes its verified

**Pull request merged**